static/js/pages/instances/InstanceSnapshots.tsx

```diff
@@ -83,45 +87,40 @@ const InstanceSnapshots: FC<Props> = ({ controlTarget, instance, notify }) => {
       {controlTarget &&
         createPortal(
           <>
-            <CreateSnapshotBtn instance={instance} notify={notify} />
+            <CreateSnapshotBtn openSnapshotForm={() => setModalOpen(true)} />
           </>,
           controlTarget
         )}
-      <Row>
-        <MainTable
-          headers={headers}
-          rows={rows}
-          paginate={30}
-          responsive
-          sortable
-          className="u-table-layout--auto"
-          emptyStateMsg={
-            <Row className="p-strip empty-state-message">
-              <Col size={4} className="u-align--right">
-                <Icon name="settings" className="u-hide--small icon" />
-              </Col>
-              <Col size={8} className="u-align--left">
-                <h4 className="p-heading--2">No snapshots found</h4>
-                <p>
-                  A snapshot is the state of an instance at a particular point
-                  in time. It can be used to restore the instance to that state.
-                  Create a new snapshot with the &quot;Create snapshot&quot;
-                  button above.
-                </p>
-                <p>
-                  <a
-                    className="p-link--external"
-                    href="https://linuxcontainers.org/lxd/docs/latest/howto/storage_backup_volume/#storage-backup-snapshots"
-                    target="_blank"
-                    rel="noreferrer"
-                  >
-                    Read the documentation on snapshots
-                  </a>
-                </p>
-              </Col>
-            </Row>
-          }
+      {isModalOpen && (
+        <CreateSnapshotForm
+          instance={instance}
+          close={() => setModalOpen(false)}
+          notify={notify}
         />
+      )}
+      <Row>
+        {instance.snapshots && instance.snapshots.length > 0 ? (
+          <MainTable
+            headers={headers}
+            rows={rows}
+            paginate={30}
+            responsive
+            sortable
+            className="u-table-layout--auto"
+          />
+        ) : (
+          <EmptyState
+            iconName="settings"
+            iconClass="p-empty-instances"
```

Comment:
In order to apply the correct colour, this should be changed to some new class called `p-empty-snapshots` which includes the mixin of the correct (temporary) icon:
```
.p-empty-snapshots {
  @include vf-icon-settings($color-mid-light);
}
```

Reply:
I changed it [here](https://github.com/canonical/lxd-ui/pull/213/files#diff-f5a5b0429dba7d86dd96396c39539ab3a091d90536bccb9db008a3dbc487f83dR102). Will pick that change to this branch, to allow reusing of the icon class.

Reply:
This is a different thing: not the icon for a link, but the icon representing the snapshots - as it was `vf-icon-containers` for instances, it is `vf-icon-settings` (temporarily) for snapshots.
It is to this icon that the correct colour (`$color-mid-light`) should be applied.

Reply:
Right, I mixed up the external link icon and the snapshot icon. Now it should be fixed.